### PR TITLE
Add NATS Streaming Protocol Documentation

### PR DESCRIPTION
**Pull request by ColinSullivan1**
This page describes the NATS Streaming protocol.  The goal is to provide enough information that one could get started building a client.

@lparis @kozlovic @aricart @mcqueary @bjflanne

### CODE REVIEW

**Comment by lparis** [comment]
LGTM, although I think it should be placed in the the NATS Streaming section (/nats-site/content/documentation/streaming) and not in /nats-site/content/documentation/internals.

File: content/documentation/internals/nats-streaming-protocol.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+| [`CloseResp`](#CLOSERESP)         | Server  | Result of the close request
+
+The following sections explain each protocol message.
+
+## <a name="CONNREQ"></a>ConnectRequest
+
+### Message Structure
+
+* `ClientID`: A unique identifier for a client.
+* `HeartbeatInbox`: An inbox to which the streaming server will send heartbeats to for the client to process.
```
Comment (aricart) [inline content/documentation/internals/nats-streaming-protocol.md]: The protocol fields (as defined in the protobuf file) use standard cammel case. This means that while in Go they  have their initial field capitalized, in other languages they are not. We should document the protocol fields in a language agnostic way.

Comment (mcqueary) [inline content/documentation/internals/nats-streaming-protocol.md]: IMO we should document them in the case that is used in the `.proto` file.

Comment (aricart) [inline content/documentation/internals/nats-streaming-protocol.md]: Looks like we do that for the other types. So the only thing incorrect here is that `ClientID` and `HeartbeatInbox` should be `clientID` and `heartbeatInbox` instead.

File: content/documentation/internals/nats-streaming-protocol.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+* `qGroup`: Optional queue group
+* `inbox`: Inbox subject to deliver messages on
+* `maxInFlight`:  Maximum inflight messages without an acknowledgement allowed
+* `ackWaitInSecs`: Timeout for receiving an acknowledgement from the client
+* `durableName`: Optional durable name which survives client restarts
+* `startPosition`: Start position type
+* `startSequence`: Optional start sequence number
+* `startTimeDelta`: Optional start time
+
+StartPosition
```
Comment (aricart) [inline content/documentation/internals/nats-streaming-protocol.md]: `StartPosition` is an enum.

File: content/documentation/internals/nats-streaming-protocol.md (hunk 1 above)
Comment (ColinSullivan1) [inline content/documentation/internals/nats-streaming-protocol.md]: Yep, this one I missed.  Thanks.  Will fix.

**Comment by aricart** [comment]
LGTM. I will look at this doc very closely when I do the unit tests for the node-nats-streaming.

File: content/documentation/internals/nats-streaming-protocol.md (hunk 2 above)
Comment (ColinSullivan1) [inline content/documentation/internals/nats-streaming-protocol.md]: Thanks, I'll add a note to that effect.

**Comment by ColinSullivan1** [comment]
@lparis I debated a bit on location.  Eventually, I expect that we'll have tutorials, client development, etc.  Should they be alongside the others, or all located in the streaming server section...  Any thoughts on the long term plan?

File: content/documentation/internals/nats-streaming-protocol.md (hunk 2 above)
Comment (mcqueary) [inline content/documentation/internals/nats-streaming-protocol.md]: It's a little hard to describe that field accurately in a way that makes sense at first glance. I would try something like this this:  

`startPosition`: An enumerated type specifying which kind of offset will be used to deliver historical data to this subscription.

**Comment by mcqueary** [comment]
Nit-picky, I realize, but all instances of "streaming server" should really be "NATS Streaming server".

**Comment by mcqueary** [comment]
I agree with Lorenzo on location. IMO no matter what the long-term plan is, in the short term all the streaming-specific pages should be in the NATS Streaming section.

**Comment by kozlovic** [comment]
LGTM.

**Comment by ColinSullivan1** [comment]
I updated the PR.  Anyone want to take another look, or are we good to go?

**Comment by kozlovic** [comment]
I had a look yesterday, LGTM.

**Comment by aricart** [comment]
LGTM

**Pull request merged**